File: java/smartlist/src/main/java/com/smartlist/module/question/questionstrategy/FileStrategy.java
package com.smartlist.module.question.questionstrategy;

import com.smartlist.module.question.QuestionResponseCodes;
import com.smartlist.module.question.dto.request.QuestionReqDTO;
import com.smartlist.module.question.interfaces.IQuestionStrategy;

public class FileStrategy implements IQuestionStrategy {

    private final QuestionReqDTO questionReqDto;

    public FileStrategy(final QuestionReqDTO questionReqDto){
        this.questionReqDto = questionReqDto;
    }

    @Override
    public void validate() {
        if (this.questionReqDto.getOptions() == null) {
            throw QuestionResponseCodes.INVALID_QUESTION_OPTION;
        }
        if (!this.questionReqDto.getAnswerOptions().isEmpty()) {
            throw QuestionResponseCodes.INVALID_ANSWER_OPTIONS_FOR_QUESTION_TYPE;
        }

        if (Boolean.TRUE.equals(!this.questionReqDto.getOptions().getFileRequired()) && Boolean.TRUE.equals(!this.questionReqDto.getOptions().getTextRequired())) {
            throw QuestionResponseCodes.INVALID_QUESTION_OPTION;
        }
    }
}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/question/question-strategy/file.strategy.ts
import { QuestionReqDto } from '../dto/request/question.req.dto';
import { IQuestionStrategy } from '../question-interface/question-interface';
import { QuestionResponseCodes } from '../question.response.codes';

export class FileStrategy implements IQuestionStrategy {
  constructor(protected questionReqDto: QuestionReqDto) {}
  validate() {
    if (this.questionReqDto.options == null) {
      throw QuestionResponseCodes.INVALID_QUESTION_OPTION;
    }
    if (this.questionReqDto.answer_options.length > 0) {
      throw QuestionResponseCodes.INVALID_ANSWER_OPTIONS_FOR_QUESTION_TYPE;
    }

    if (
      !this.questionReqDto.options.file_required &&
      !this.questionReqDto.options.text_required
    ) {
      throw QuestionResponseCodes.INVALID_QUESTION_OPTION;
    }
  }
}
